samtranslator/model/api/api_generator.py

```diff
@@ -326,7 +326,9 @@ def _add_auth(self):
             self._set_default_apikey_required(swagger_editor)
 
         if auth_properties.ResourcePolicy:
-            swagger_editor.add_resource_policy(auth_properties.ResourcePolicy)
+            for path in swagger_editor.iter_on_path():
+                swagger_editor.add_resource_policy(auth_properties.ResourcePolicy, path,
```

Comment:
This is for the case where a resource policy is added to a function event, correct? This particular path needs to be able to have a policy on a particular path and not apply to the entire API.